test/tests/p5.SoundRecorder.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+define(['chai'], function(chai) {
+  var expect = chai.expect;
+
+  describe('p5.SoundRecorder', function() {
+    var recorder;
+    var inputSoundFile;
+    var originalWriteFile;
+    var writeFileCalled;
+
+    before(function(done) {
+      // clean up audio nodes from other test suites
+      p5.soundOut.soundArray.length = 0;
+
+      // mock p5.prototype.writeFile
+      originalWriteFile = p5.prototype.writeFile;
+      p5.prototype.writeFile = function() {
+        writeFileCalled = true;
+      };
+
+      inputSoundFile = p5.prototype.loadSound('./testAudio/constant.wav', function() {
+        done();
+      });
+    });
+
+    after(function() {
+      inputSoundFile.dispose();
+      p5.prototype.writeFile = originalWriteFile;
+    });
+
+    beforeEach(function() {
+      recorder = new p5.SoundRecorder();
+      inputSoundFile.disconnect();
+      inputSoundFile.stop();
+      writeFileCalled = false;
+    });
+
+    afterEach(function() {
+      recorder.dispose();
+    });
+
+    it('can record input from a microphone', function(done) {
+      var recordingDuration = 100;
+
+      // need to enable master volume to test recording from the microphone
+      p5.prototype.masterVolume(1);
+
+      var mic = new p5.AudioIn();
```

Comment:
we discussed adding an `await` for mic access if we return the promise here

https://github.com/processing/p5.js-sound/blob/master/src/audioin.js#L133

Reply:
after giving it some thought, I'd rather not modify `p5.AudioIn` as part of this PR. instead, I'll add a request for mic access to the `before` hook for this test suite, and throw an error if mic access is denied

Reply:
👍